scripts/train_agents.py

```diff
@@ -12,17 +16,55 @@ def SP(args, pop_force_training):
         'generate': [TeamType.SELF_PLAY],
         'load': get_eval_types_to_load()
     }
-    get_selfplay_agent(args=args,
-                       train_types=args.sp_train_types,
-                       eval_types=args.sp_eval_types,
-                       total_training_timesteps=args.pop_total_training_timesteps,
-                       force_training=pop_force_training,
-                        )
+    _, _ = get_selfplay_agent_w_tms_collection(args=args,
+                                            train_types=args.sp_train_types,
+                                            eval_types=args.sp_eval_types,
+                                            total_training_timesteps=args.pop_total_training_timesteps,
+                                            force_training=pop_force_training)
+
+
+def SP_w_SP_Types(args, pop_force_training:bool, parallel:bool) -> None:
+    '''
+    Set up and run the training for self-play with self-play types
+    Similar to FCP_w_SP_TYPES, this function will first train a population of SP agents, organize them into a teammates_collection
+    based on TeamType, and then select agents from th the teams to SP with a randomly initialized SP agent
+    So the randomly initialized agent will train with itself and one other unseen teammate (e.g. [SP, SP, SP, SP_H] in a 4-chef layout)
+    
+    :param pop_force_training: Boolean that, if true, indicates population should be generated, otherwise load it from file
+    :param parallel: Boolean indicating if parallel envs should be used for training or not
+    '''
+
+    # Set the train and eval types that will be used to organize the first "FCP-like" population
+    # NOTE: When in order to use SELF_PLAY_X later, we need X_FIRST here
+    args.sp_train_types = [TeamType.HIGH_FIRST]
+    args.sp_eval_types = {
+        'generate': [TeamType.HIGH_FIRST],
+        'load': []
+        }
+
+    # Set the train and eval types for the teammates that will be used to play with the SP agent
```

Comment:
I think when you make the changes in my prev comment, you can also get rid of this comment